cmd/eval.go

```diff
@@ -83,9 +85,78 @@ func evalCmd() *cobra.Command {
 			return nil
 		},
 		RunE: func(cmd *cobra.Command, args []string) error {
-			//TODO: refactor once we handle application snapshots
-			if arguments.input != "" || arguments.filepath != "" {
-				report, err := applicationsnapshot.Report(snapshotSpec)
+			validateImage := func(imageRef string) (*policy.Output, error) {
+				out := &policy.Output{}
+
+				i, err := image.NewImageValidator(cmd.Context(), imageRef, arguments.publicKey, arguments.rekorURL)
+				if err != nil {
+					return nil, err
+				}
+
+				if err := i.ValidateImageSignature(cmd.Context()); err != nil {
+					out.SetImageSignatureCheck(false, err.Error())
+					return nil, err
+				}
+				out.SetImageSignatureCheck(true, "success")
+
+				if err := i.ValidateAttestationSignature(cmd.Context()); err != nil {
+					out.SetAttestationSignatureCheck(false, err.Error())
+					return nil, err
+				}
+				out.SetAttestationSignatureCheck(true, "success")
+
+				p, err := policy.NewPolicyEvaluator(arguments.policyConfiguration)
+				if err != nil {
+					return nil, err
+				}
+
+				results, err := p.Evaluate(cmd.Context(), i.Attestations())
+				if err != nil {
+					return nil, err
+				}
+				out.SetPolicyCheck(results)
+
+				return out, nil
+			}
+
+			if snapshotSpec != nil {
```

Comment:
Something to refactor in a followup PR, we could create an artificial ApplicationSnapshot when passing the image via `--image` and here assume we were given ApplicationSnapshot